src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -109,17 +148,43 @@ void Bmi_Multi_Formulation::create_multi_formulation(geojson::PropertyMap proper
  * @return Whether a model may perform updates beyond its ``end_time``.
  */
 const bool &Bmi_Multi_Formulation::get_allow_model_exceed_end_time() const {
-    for (const std::shared_ptr<Bmi_Formulation>& m : modules) {
+    for (const std::shared_ptr<Bmi_Module_Formulation<bmi::Bmi>>& m : modules) {
         if (!m->get_allow_model_exceed_end_time())
             return m->get_allow_model_exceed_end_time();
     }
     return modules.back()->get_allow_model_exceed_end_time();
 }
 
+/**
+ * Get the collection of forcing output property names this instance can provide.
+ *
+ * For this type, this is the collection of the names/aliases of the BMI output variables for nested modules;
+ * i.e., the config-mapped alias for the variable when set in the realization config, or just the name when no
+ * alias was included in the configuration.
+ *
+ * This is part of the @ref ForcingProvider interface.  This interface must be implemented for items of this
+ * type to be usable as "forcing" providers for situations when some other object needs to receive as an input
+ * (i.e., one of its forcings) a data property output from this object.
+ *
+ * @return The collection of forcing output property names this instance can provide.
+ * @see ForcingProvider
+ */
+const vector<std::string> &Bmi_Multi_Formulation::get_available_forcing_outputs() {
```

Comment:
Could they also provide forcing from the "ForcingProvider", e.g. AORC variables at the top of the provider "stack"

Reply:
@hellkite500, @mattw-nws, I think I see where you both are going with this now ... the copying is excessive.  I still don't feel great about that kind of transitive providing, but I'm sure there is a better alternative than what it is doing now.